datasets/covid19_tracking/state_screenshots/custom/download_screenshots.py

```diff
@@ -17,42 +17,55 @@
 import os
 import pathlib
 import typing
+from multiprocessing.pool import ThreadPool
 
 import requests
-from google.cloud import storage
 
 
-def main(csv_path: pathlib.Path, source_column: str, target_column: str):
-    gcs_client = storage.Client()
+def download_item(source_target: typing.Tuple[str, pathlib.Path]):
+    source_url, download_path = source_target
+    if not os.path.exists(download_path):
+        r = requests.get(source_url, stream=True)
+        if r.status_code == 200:
+            with open(download_path, "wb") as f:
+                for chunk in r:
+                    f.write(chunk)
+
+
+def download_parallel(source_targets: typing.List[typing.Tuple[str, pathlib.Path]]):
+    ThreadPool(10).imap_unordered(download_item, source_targets)
```

Comment:
10 is a [magic number](https://wiki.c2.com/?MagicNumber). At the very least, we should have a constant for this value.

Also, how do we know that 10 is the right value? Are we trying to avoid having too much load on the server? Are we trying to optimize for the number of CPUs we have available? We should document why 10 was chosen.

Reply:
As discussed on chat, we can find the number of CPUs via `os.cpu_count()` and use half (0.5) of that as a fair estimate on the number of worker threads we'd like to use.

Please see https://github.com/GoogleCloudPlatform/public-datasets-pipelines/pull/3/commits/eb12e90242ed9571ece2359b4c7801df03d16f9d.